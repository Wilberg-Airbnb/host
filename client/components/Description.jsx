import React from 'react';
import { Item } from 'semantic-ui-react';
import Superhost from './Superhost.jsx';
import DuringYourStay from './DuringYourStay.jsx';



const Description = ({description, superhost, name, duringYourStay}) => (
  <Item.Group>
    <Item>
      <Item.Content>
        <Item.Description>
        {description}
        </Item.Description>
      </Item.Content>
    </Item>
    {duringYourStay !== null ? <DuringYourStay duringYourStay={duringYourStay}/> : null}
    {superhost !== null ? <Superhost name={name}/> : null}
  </Item.Group>

)

export default Description;