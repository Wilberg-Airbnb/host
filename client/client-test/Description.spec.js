import React from 'react';
import Description from '../components/Description';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
configure({ adapter: new Adapter() });
 
describe('Renders Components', () => {
  let component;
  let props = {
    description: 'Explicabo nisi dolorum. Accusantium quam eum sint voluptas molestiae ullam ad. Ipsa est illo doloremque itaque. Excepturi quia reprehenderit qui ex architecto.',
    superhost: null,
    duringYourStay: null,
    name: 'Marg Rob'
  }
  beforeEach( () => {
    component = shallow(<Description description={props.description} superhost={props.superhost} duringYourStay={props.duringYourStay} name={props.name}/>)
  });

  test("should render initial layout", () => {
    const component = shallow(<Description description={props.description} superhost={props.superhost} duringYourStay={props.duringYourStay} name={props.name} />);
    expect(component.getElements()).toMatchSnapshot();
  });

});


