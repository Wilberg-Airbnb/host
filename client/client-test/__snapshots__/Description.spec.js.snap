// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Components should render initial layout 1`] = `
Array [
  <ItemGroup>
    <Item>
      <ItemContent>
        <ItemDescription>
          Explicabo nisi dolorum. Accusantium quam eum sint voluptas molestiae ullam ad. Ipsa est illo doloremque itaque. Excepturi quia reprehenderit qui ex architecto.
        </ItemDescription>
      </ItemContent>
    </Item>
  </ItemGroup>,
]
`;
